lib/rouge/lexers/plsql.rb

```diff
@@ -0,0 +1,596 @@
+        #
+        rule %r/[+-]?(?:(?:\.\d+(?:[eE][+-]?\d+)?)|\d+\.(?:\d+(?:[eE][+-]?\d+)?))/, Num::Float
+        rule %r/[+-]?\d+/, Num::Integer
+        
+        # 
+        # Operators
+        #
+        # Special semi-operator, but this seems an appropriate classification
+        rule %r/%(?:TYPE|ROWTYPE)\b/i, Name::Attribute
+        # longer ones come first on purpose! It matters to regex engine
+        rule %r/=>|\|\||\*\*|<<|>>|\.\.|<>|[:!~^<>]=|[-+%\/*=<>@&!^\[\]]/, Operator
+        rule %r/(NOT|AND|OR|LIKE|BETWEEN|IN)(\s)/im do
+            groups Operator::Word, Text
+        end
+        rule %r/(IS)(\s+)(?:(NOT)(\s+))?(NULL\b)/im do
+            groups Operator::Word, Text, Operator::Word, Text, Operator::Word
+        end
+
+        # 
+        # Punctuation
+        #
+        # special case of dot followed by a name. notice the lookahead assertion
+        rule %r/\.(?=\w)/ do
+            token Punctuation
+            push :dotnames
+        end
+        rule %r/[;:()\[\],.]/, Punctuation
+
+        #
+        # Special processing for keywords with multiple contexts
+        #
+        # this madness is to keep the word "replace" from being treated as a builtin function in this context
+        rule %r/(create)(\s+)(?:(or)(\s+)(replace)(\s+))?(package|function|procedure|type)(?:(\s+)(body))?(\s+)(\w[\w\d\$]*)/im do
+            groups Keyword::Reserved, Text, Keyword::Reserved, Text, Keyword::Reserved, Text, Keyword::Reserved, Text, Keyword::Reserved, Text, Name
+        end
+        # similar for MERGE keywords
+        rule %r/(when)(\s+)(?:(not)(\s+))?(matched)(\s+)(then)(\s+)(update|insert)\b(?:(\s+)(set)(\s+))?/im do
+            groups Keyword::Reserved, Text, Keyword::Reserved, Text, Keyword::Reserved, Text, Keyword::Reserved, Text, Keyword::Reserved, Text, Keyword::Reserved, Text
+        end
+
+        #
+        # General keyword classification with sepcial attention to names
+        # in a chained "dot" notation.
+        #
+        rule %r/(\w[\w\d\$]*)(\.(?=\w))?/ do |m|
+          if self.class.keywords_type.include? m[1].upcase
+            tok = Keyword::Type 
+            #Name::Builtin
+          elsif self.class.keywords_func.include? m[1].upcase
+            tok = Name::Function
+          elsif self.class.keywords.include? m[1].upcase
+            tok = Keyword::Reserved
+          elsif self.class.keywords_nresvd.include? m[1].upcase
+            tok = Keyword
+          else
+            tok = Name
+          end
+          groups tok, Punctuation
+
+          if m[2] == "."
+            push :dotnames
+          end
+        end
+      end
+
+      state :multiline_comments do
+        rule %r/([*][^\/]|[^*])+/m, Comment::Multiline
+        rule %r([*]\/), Comment::Multiline, :pop!
+      end
+
+
+      #state :backtick do
+      #  rule %r/\\./, Str::Escape
+      #  rule %r/``/, Str::Escape
+      #  rule %r/`/, Name::Variable, :pop!
+      #  rule %r/[^\\`]+/, Name::Variable
+      #end
```

Comment:
**suggestion:** If backtick is not supported in PL/SQL, can we remove this block altogether instead of commenting?

Reply:
Thanks for the detailed review Tan!
I'll address all of this as time permits. It shouldn't take long (though not today).
Unfortunately, if I dive back in I'll be tempted to tweak some more. I'll try to restrain myself.